File: src/main/java/ac/dnd/server/admission/infrastructure/persistence/repository/EventJpaRepository.java
package ac.dnd.server.admission.infrastructure.persistence.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;

import ac.dnd.server.admission.infrastructure.persistence.entity.Event;
import ac.dnd.server.common.support.EventStatus;

public interface EventJpaRepository extends JpaRepository<Event, Long> {

	List<Event> findByStatusIn(List<EventStatus> statuses);

}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/persistence/repository/EventJpaRepository.kt
package ac.dnd.server.admission.infrastructure.persistence.repository

import ac.dnd.server.admission.infrastructure.persistence.entity.EventEntity
import ac.dnd.server.admission.domain.enums.EventStatus
import org.springframework.data.jpa.repository.JpaRepository

interface EventJpaRepository : JpaRepository<EventEntity, Long> {

    fun findByStatusIn(statuses: List<EventStatus>): List<EventEntity>
}
